controllers/idler/aap_idler_test.go

```diff
@@ -0,0 +1,563 @@
+
+	t.Run("with running AAPs, nothing to idle", func(t *testing.T) {
+		// given
+		aapIdler, interceptedNotify := prepareAAPIdler(t, idler, idledAAP, runningAAP, runningNoSpecAAP, noiseAAP)
+
+		isOwningSomething := false
+		preparePayloadsForAAPIdler(t, aapIdler, func(kind schema.GroupVersionKind, object client.Object) {
+			if kind.Kind == "Deployment" {
+				if !isOwningSomething {
+					require.NoError(t, controllerutil.SetOwnerReference(runningAAP, object, scheme.Scheme))
+					isOwningSomething = true
+				}
+			}
+		}, idler.Name, "short-", freshStartTimes(aapTimeoutSeconds(idler.Spec.TimeoutSeconds)))
+
+		preparePayloadCrashloopingPodsWithinThreshold(t, clientSetForIdler(t, aapIdler, func(kind schema.GroupVersionKind, object client.Object) {
+			require.NoError(t, controllerutil.SetOwnerReference(runningNoSpecAAP, object, scheme.Scheme))
+		}), idler.Name, "restarting-", freshStartTimes(aapTimeoutSeconds(idler.Spec.TimeoutSeconds)))
+
+		// when
+		requeueAfter, err := aapIdler.ensureAnsiblePlatformIdling(context.TODO(), idler)
+
+		// then
+		require.NoError(t, err)
+		assert.Greater(t, requeueAfter, (time.Duration(aapTimeoutSeconds(idler.Spec.TimeoutSeconds)/2)-1)*time.Second)
+		assert.Less(t, requeueAfter, (time.Duration(aapTimeoutSeconds(idler.Spec.TimeoutSeconds)/2)+1)*time.Second)
+		interceptedNotify.assertThatCalled(t)
+		assertAAPsIdled(t, aapIdler, idler.Name, idledAAP.GetName())
+	})
+
+	t.Run("with running AAPs, one idled, second scheduled", func(t *testing.T) {
+		// given
+		aapIdler, interceptedNotify := prepareAAPIdler(t, idler, idledAAP, runningAAP, runningNoSpecAAP, noiseAAP)
+		isOwningSomething := false
+		preparePayloadsForAAPIdler(t, aapIdler, func(kind schema.GroupVersionKind, object client.Object) {
+			if kind.Kind == "Deployment" {
+				if !isOwningSomething {
+					require.NoError(t, controllerutil.SetOwnerReference(runningAAP, object, scheme.Scheme))
+					isOwningSomething = true
+				}
+			}
+		}, idler.Name, "long-", expiredStartTimes(aapTimeoutSeconds(idler.Spec.TimeoutSeconds)))
+		preparePayloadsForAAPIdler(t, aapIdler, func(kind schema.GroupVersionKind, object client.Object) {
+			if kind.Kind == "ReplicaSet" && len(object.GetOwnerReferences()) == 0 {
+				require.NoError(t, controllerutil.SetOwnerReference(runningNoSpecAAP, object, scheme.Scheme))
+			}
+		}, idler.Name, "crashlooping-", freshStartTimes(aapTimeoutSeconds(idler.Spec.TimeoutSeconds)))
+
+		// when
+		requeueAfter, err := aapIdler.ensureAnsiblePlatformIdling(context.TODO(), idler)
+
+		// then
+		require.NoError(t, err)
+		assert.Greater(t, requeueAfter, (time.Duration(aapTimeoutSeconds(idler.Spec.TimeoutSeconds)/2)-1)*time.Second)
+		assert.Less(t, requeueAfter, (time.Duration(aapTimeoutSeconds(idler.Spec.TimeoutSeconds)/2)+1)*time.Second)
+		interceptedNotify.assertThatCalled(t, runningAAP.GetName())
+		assertAAPsIdled(t, aapIdler, idler.Name, idledAAP.GetName(), runningAAP.GetName())
+	})
+
+	t.Run("all idled", func(t *testing.T) {
```

Comment:
Maybe I missed it but you always idle one (or none) AAPs in your tests, right? Can we add another AAP to idle to make sure we idle two/multiple (or none)? To test the case when there is more than one AAP to idle.

Reply:
This test was actually supposed to idle the long-running AAP as well as the crashlooping one (so two AAP instances), but there was a bug. Fixed in: [3761ed5](https://github.com/codeready-toolchain/member-operator/pull/631/commits/3761ed51e26c532e3eb6076599a1fd8d8773a5d3)